### PR TITLE
AN-207 Add support for gzipped VCFs in IGV

### PR DESCRIPTION
**Pull request by aednichols**
Objective: Allow Eric Weitz to use Terra web IGV with the files he wants, ending in `vcf.gz` plus `.vcf.gz.tbi` index.

- Allow `.gz` file extension
- Map `.gz` file to `.gz.tbi` index
  - It's possible that there are `.gz` files requiring different index extensions, but I consider that out of scope

Tested by loading up the files Beth provided in my [test workspace ](https://bvdp-saturn-dev.appspot.com/#workspaces/testing-job-manager/lots_of_workflows/data) and viewing the file. Terra now recognizes the combination and allows IGV.

I don't know enough about IGV to evaluate its behavior, but it certainly seems to be working.

Screenshot 2024-11-06 at 17 51 32

### CODE REVIEW

**Review by lucymcnatt — APPROVED** [review]

File: src/components/IGVFileSelector.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ const isTdrUrl = (fileUrl) => {
 };
 
 const findIndexForFile = (fileUrl, fileUrls) => {
-  if (!['.cram', '.bam', '.vcf'].some((extension) => fileUrl.endsWith(extension))) {
+  if (!['.cram', '.bam', '.vcf', '.gz'].some((extension) => fileUrl.endsWith(extension))) {
```
Comment (pshapiro4broad) [inline src/components/IGVFileSelector.js]: I am not an IGV expert but `.gz` seems too permissive to me, as I think the only gzip files that are supported are vcfs. Could this use `.vcf.gz` instead?

See https://igv.org/doc/desktop/#FileFormats/DataTracks/#vcf

Comment (LizBaldo) [reply src/components/IGVFileSelector.js]: It looks like `gz` is mapped to `.gz.tbi` below? But that is a good question, what would happen if a user tries to load a different type of compressed file that does not untar to .tbi?

Comment (aednichols) [reply src/components/IGVFileSelector.js]: I agree with you but it's annoying with how the code is structured right now.

`getValidIgvFiles` uses `splitExtension` which considers only the last segment `.gz` to be the extension.

Meanwhile, we have a duplicative allowlist in `findIndexForFile` that applies `endsWith` and matches `.vcf.gz`.

I'll poke at it some more and see if I can improve things.

Comment (aednichols) [reply src/components/IGVFileSelector.js]: > what would happen if a user tries to load a different type of compressed file that does not untar to .tbi

Nothing bad, the line `gz: [${base}.gz.tbi]` does not match and we do not enable IGV.

Comment (LizBaldo) [reply src/components/IGVFileSelector.js]: Gotcha, in that case I am not supper worried about adding `gz` as is honestly

Comment (eweitz) [reply src/components/IGVFileSelector.js]: I also don't see a practical issue with adding `.gz` as is, but only adding support for `.vcf.gz` as Phil suggests would in principle suffice for this use case.  BAM and CRAM essentially never appear with `.gz` suffixes.  BAM is gzipped (via bgzip) by definition, making the `.gz` suffix superfluous.

IGV supports other genomics data files that appear with and without a .gz suffix.  [BED files](https://genome.ucsc.edu/FAQ/FAQformat.html) are another common genomics data format that often appears as `${base}.gz`, e.g. `foo.bed.gz`.  These often pair with a TBI file, e.g. `foo.bed.gz.tbi`.  If you open Chrome DevTools, go to Network panel, then go to the public study [SCP2699 in Single Cell Portal](https://singlecell.broadinstitute.org/single_cell/study/SCP2699/multiome-demo-study-10x-cell-ranger-arc-processed?genes=PTEN&tab=genome#study-visualize), you can see an example `.bed.gz` and `.bed.gz.tbi` pair being fetched to render the igv.js track with grey and red features.

[Others genomic file formats](https://genome.ucsc.edu/FAQ/FAQformat.html) viewable in IGV that are slightly less common but foreseeable include GFF / GFF3 and GTF with and without `.gz` suffixes, e.g. `foo.gff`, `foo.gff.gz`, `foo.gff3`, `foo.gff3.gz`, `foo.gtf`, `foo.gtf.gz`.  

Those other file extensions aren't necessarily in scope here, but hopefully the extra context helps.

**Review by eweitz — APPROVED** [review]
Code looks good, and I manually confirmed this works!

File: src/components/IGVFileSelector.js (hunk 1 above)
Comment (aednichols) [reply src/components/IGVFileSelector.js]: Eric, thank you for taking the time to test the PR and leave the expert comment. It provides useful domain info that will help us with the long-term maintenance of this code.

**Review by lucymcnatt — APPROVED** [review]
changes make sense and tested successfully, 👍

**Pull request merged**